internal/ark/config.go

```diff
@@ -272,9 +363,20 @@ func (req ConfigRequest) getCredentials() (credentials, error) {
 
 	return credentials{
 		SecretContents: secretContents{
-			Secret:  azureSecret,
 			Cluster: ClusterSecretContents,
-			Bucket:  BucketSecretContents,
+			Cloud:   BucketSecretContents,
 		},
 	}, err
 }
+
+func getPullPolicy(pullPolicy string) v1.PullPolicy {
+	switch pullPolicy {
+	case string(v1.PullAlways):
+		return v1.PullAlways
+
+	case string(v1.PullNever):
+		return v1.PullNever
+	default:
+		return v1.PullIfNotPresent
+	}
+}
```

Comment:
**Optional**: opinionated, but this **MAY** be easier to read/maintain the following way:

```suggestion
func getPullPolicy(pullPolicy string) v1.PullPolicy {
	switch pullPolicy {
	case string(v1.PullAlways), string(v1.PullIfNotPresent), string(v1.PullNever): // Note: known values.
		return v1.PullPolicy(pullPolicy)
	default:
		return v1.PullIfNotPresent
	}
}
```